src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using GuardNet;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Arcus.Messaging.Pumps.Abstractions.MessageHandling
```

Comment:
Can we provide tests for our voodoo part? ⚔😵

Reply:
We should really test the outside of this. Like, registering several message handlers with different messages and see if we can select them all.